src/hooks/useFileUpload.tsx

```diff
@@ -97,6 +101,40 @@ export default function useFileUpload(
   const [files, setFiles] = useState<File[]>([]);
   const queryClient = useQueryClient();
 
+  const generatePDF = async (files: File[]): Promise<File | null> => {
+    try {
+      const pdf = new jsPDF();
+      for (const [index, file] of files.entries()) {
+        if (!file.type.startsWith("image/")) {
+          console.error(`Unsupported file type: ${file.name}`);
+          toast.error(t("file_error__file_type"));
+          return null;
+        }
+        const imgData = await new Promise<string>((resolve, reject) => {
+          const reader = new FileReader();
+          reader.onload = () => resolve(reader.result as string);
```

Comment:
This maybe unnecessary, but play around with larger number of image files (like say 10-15) and see how quickly it converts to pdf.

If it does take a few seconds, might be worth using the progress bar here.

You can use setProgress for it (and reset it to 0 once pdf file is created as it's also used for indicating file upload progress). You would also want to inform the user file is being converted through toast.

Experiment and check if it fits within existing UI, if not we can discuss further.